src/main/kotlin/com/example/delivery/model/MongoOrder.kt

```diff
@@ -0,0 +1,36 @@
+package com.example.delivery.model
+
+import com.example.delivery.dto.response.OrderDTO
+import org.springframework.data.annotation.Id
+import org.springframework.data.annotation.TypeAlias
+import org.springframework.data.mongodb.core.mapping.Document
+
+@TypeAlias("Order")
+@Document(collection = MongoOrder.COLLECTION_NAME)
+data class MongoOrder(
+    @Id val id: String? = null,
```

Comment:
Use ObjectId as Id always. Do not build any logic around this Id. If you want to provide any identifier to the customer create separate field for it